src/OrchardCore.Modules/OrchardCore.ContentTypes/Services/ContentFieldLocalizationDataProvider.cs

```diff
@@ -0,0 +1,23 @@
+using System.Collections.Generic;
+using System.Linq;
+using OrchardCore.Localization;
+
+namespace OrchardCore.ContentTypes.Services
+{
+    public class ContentFieldDataLocalizationProvider : ILocalizationDataProvider
+    {
+        private readonly IContentDefinitionService _contentDefinitionService;
+
+        public ContentFieldDataLocalizationProvider(IContentDefinitionService contentDefinitionService)
+        {
+            _contentDefinitionService = contentDefinitionService;
+        }
+        
+        // TODO: Check if there's a better way to get the fields
+        public IEnumerable<string> GetAllStrings()
+            => _contentDefinitionService.GetTypes()
+                    .SelectMany(t => t.TypeDefinition.Parts)
+                    .Where(p => p.PartDefinition.Fields.Count() > 0)
```

Comment:
This line is unnecessary as the next one will just return nothing if count is `0`

Reply:
I will check it again